### PR TITLE
Avoid unnecessary locking on runtimeImplMap

### PR DESCRIPTION
**Pull request by tedyu**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:
In Runtime#DeleteContainer, when the impl was not found in the runtimeImplMap, there is no need to take write lock on the map at the end of the method.
Assuming the get from runtimeImplMap at beginning of DeleteContainer() returns nothing, the deferred func shouldn't remove map entry someone else puts in the map.

#### Which issue(s) this PR fixes:

```release-note
None
```

### CODE REVIEW

File: internal/oci/oci.go [hunk 1]
```diff
@@ -302,19 +302,21 @@ func (r *Runtime) StopContainer(ctx context.Context, c *Container, timeout int64
...
 // DeleteContainer deletes a container.
 func (r *Runtime) DeleteContainer(c *Container) (err error) {
-	impl, err := r.RuntimeImpl(c)
-	if err != nil {
-		return err
+	r.runtimeImplMapMutex.RLock()
+	impl, ok := r.runtimeImplMap[c.ID()]
+	r.runtimeImplMapMutex.RUnlock()
+	if !ok {
+		impl, _ = r.newRuntimeImpl(c)
```
Comment (saschagrunert) [inline internal/oci/oci.go:309]: Hm, what are we going to do with that error?

Comment (tedyu) [reply internal/oci/oci.go]: Check added on the error return.

**Pull request merged**